mongodbatlas/fw_resource_mongodbatlas_search_deployment.go

```diff
@@ -280,22 +298,22 @@ func waitSearchNodeStateTransition(ctx context.Context, projectID, clusterName s
 	return nil, errors.New("did not obtain valid result when waiting for search deployment state transition")
 }
 
-func waitSearchNodeDelete(ctx context.Context, projectID, clusterName string, client *admin.APIClient, timeout time.Duration) error {
+func waitSearchNodeDelete(ctx context.Context, projectID, clusterName string, client SearchDeploymentService, timeConfig retrystrategy.TimeConfig) error {
 	stateConf := &retry.StateChangeConf{
 		Pending:    []string{retrystrategy.RetryStrategyIdleState, retrystrategy.RetryStrategyUpdatingState, retrystrategy.RetryStrategyPausedState},
 		Target:     []string{retrystrategy.RetryStrategyDeletedState},
 		Refresh:    searchDeploymentRefreshFunc(ctx, projectID, clusterName, client),
-		Timeout:    timeout,
-		MinTimeout: 30 * time.Second,
-		Delay:      1 * time.Minute,
+		Timeout:    timeConfig.Timeout,
+		MinTimeout: timeConfig.MinTimeout,
+		Delay:      timeConfig.Delay,
 	}
 	_, err := stateConf.WaitForStateContext(ctx)
 	return err
 }
 
-func searchDeploymentRefreshFunc(ctx context.Context, projectID, clusterName string, client *admin.APIClient) retry.StateRefreshFunc {
+func searchDeploymentRefreshFunc(ctx context.Context, projectID, clusterName string, client SearchDeploymentService) retry.StateRefreshFunc {
```

Comment:
let's keep using reference here
```suggestion
func searchDeploymentRefreshFunc(ctx context.Context, projectID, clusterName string, client *SearchDeploymentService) retry.StateRefreshFunc {
```

Reply:
SearchDeploymentService is an interface type, which in turn is a pointer to struct that fulfills the interface (instantiated in `NewSearchDeploymentServiceFromClient`). More details of why pointer to interfaces is discouraged: https://stackoverflow.com/a/44372954